astropy/io/fits/tests/test_diff.py

```diff
@@ -456,6 +469,16 @@ def test_different_table_data(self):
         assert diff.diff_total == 13
         assert diff.diff_ratio == 0.65
 
+        report = diff.report()
+        assert ('Column A data differs in row 0:\n'
+                '    a> True\n'
+                '    b> False') in report
+        assert ('...and at 13 more indices.\n'
+                ' Column D data differs in row 0:') in report
+        assert ('13 different table data element(s) found (65.00% different)'
+                in report)
+        assert len(re.findall('more indices', report)) == 1
```

Comment:
I still can't follow you. But maybe I'm misunderstanding something.

Ì thought `report` is a string and strings have a [`count`](https://docs.python.org/3/library/stdtypes.html#str.count) method that counts non-overlapping occurences of a substring. So it's completly identical to `re.findall` because you just use a string and not a regex as `substring` argument:

```
>>> diff.report().count('more indices')
1
>>> len(re.findall('more indices', report))
1
```